aspnetcore/fundamentals/servers/kestrel/options.md

```diff
@@ -276,10 +276,27 @@ Kestrel can be configured to send HTTP/2 pings to connected clients. HTTP/2 ping
 * Keep idle connections alive. Some clients and proxy servers close connections that are idle. HTTP/2 pings are considered as activity on a connection and prevent the connection from being closed as idle.
 * Close unhealthy connections. Connections where the client doesn't respond to the keep alive ping in the configured time are closed by the server.
 
-There are two configuration options related to HTTP/2 keep alive pings:
+There are two configuration options related to [HTTP/2](xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerLimits.Http2) keep alive pings:
 
-* `Http2.KeepAlivePingInterval` is a `TimeSpan` that configures the ping interval. The server sends a keep alive ping to the client if it doesn't receive any frames for this period of time. Keep alive pings are disabled when this option is set to `TimeSpan.MaxValue`. The default value is `TimeSpan.MaxValue`.
-* `Http2.KeepAlivePingTimeout` is a `TimeSpan` that configures the ping timeout. If the server doesn't receive any frames, such as a response ping, during this timeout then the connection is closed. Keep alive timeout is disabled when this option is set to `TimeSpan.MaxValue`. The default value is 20 seconds.
+::: moniker range=">= aspnetcore-5.0"
+
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Http2Limits.KeepAlivePingDelay?displayProperty=nameWithType> is a <xref:System.TimeSpan> that configures the ping interval. The server sends a keep alive ping to the client if it doesn't receive any frames for this period of time. Keep alive pings are disabled when this option is set to <xref:System.TimeSpan.MaxValue?displayProperty=nameWithType>. The default value is <xref:System.TimeSpan.MaxValue?displayProperty=nameWithType>.
+* <xref:Microsoft.AspNetCore.Server.Kestrel.Core.Http2Limits.KeepAlivePingTimeout?displayProperty=nameWithType> is a <xref:System.TimeSpan> that configures the ping timeout. If the server doesn't receive any frames, such as a response ping, during this timeout then the connection is closed. Keep alive timeout is disabled when this option is set to <xref:System.TimeSpan.MaxValue?displayProperty=nameWithType>. The default value is 20 seconds.
+
+```csharp
+webBuilder.ConfigureKestrel(serverOptions =>
+{
+    serverOptions.Limits.Http2.KeepAlivePingDelay = TimeSpan.FromSeconds(30);
+    serverOptions.Limits.Http2.KeepAlivePingTimeout = TimeSpan.FromSeconds(60);
+});
+```
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-5.0"
```

Comment:
`Http2.KeepAlive*` property was introduced in ASP.NET Core 5.0 (.NET 5) and does not exist before that. (https://github.com/dotnet/aspnetcore/issues/15104, https://github.com/dotnet/aspnetcore/pull/22565)

This section is not necessary for earlier versions.

Reply:
Thanks ... I was just about to ask James. I'll revert that bit in a second. I need to ask James about something else.